voter-dapp/src/containers/VoteData.js

```diff
@@ -0,0 +1,121 @@
+import { createContainer } from "unstated-next";
+import { useState, useEffect } from "react";
+import { drizzleReactHooks } from "@umaprotocol/react-plugin";
+
+import { useQuery } from "@apollo/client";
+import { PRICE_REQUEST_VOTING_DATA } from "../apollo/queries";
+
+// Retrieve vote data per price request from graphQL API.
+function useVoteData() {
+  const { drizzle } = drizzleReactHooks.useDrizzle();
+  const VotingContract = drizzle.contracts.Voting;
+  const { web3 } = drizzle;
+  const { toBN, fromWei, toWei, toChecksumAddress } = web3.utils;
+  const [roundVoteData, setRoundVoteData] = useState({});
+
+  // Because apollo caches results of queries, we will poll/refresh this query periodically.
+  // We set the poll interval to a very slow 5 seconds for now since the position states
+  // are not expected to change much.
+  // Source: https://www.apollographql.com/docs/react/data/queries/#polling
+  const { loading, error, data } = useQuery(PRICE_REQUEST_VOTING_DATA, {
+    pollInterval: 5000
+  });
+
+  const getRequestKey = (time, identifier, roundId) => {
+    return identifier + "-" + time + "-" + roundId;
+  };
+  const getVoteStats = () => {
+    if (error) {
+      console.error("Failed to get data:", error);
+    }
+    if (!loading && data) {
+      const newVoteData = {};
+
+      data.priceRequestRounds.forEach(async dataForRequest => {
```

Comment:
Does `forEach` accept async functions? Or does it just call them like regular functions and expect that they'll finish in the bg?

Reply:
Good catch I'll implement it like this: https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

Reply:
Makes sense +1.